### DISCUSSION TITLE
Table row click

### DISCUSSION THREAD

**Discussion by ghostdevv** [Budibase/budibase#1723] [Help]
Hello again :smile: 
How would I make it so I can redirect the user when the click on a row in the table, I essentially want it so that they click on the row and it brings up the view page for that row

**Comment by joebudi** [comment]
Hey

- Add a button to your table
- Click define actions
- Select add action - navigate to
- Change the navigate URL to something like - /applications/{{ New Table.Applicants._id }}
_the {{ New Table.Applicants._id }} part is obtained by clicking the little lightning bolt/binding icon, besides where you enter the URL, and selecting the id option_

I hope that helps.

**Reply by ghostdevv** [reply]
Is it not possible to add a click to a row instead of a button?

**Reply by shogunpurple** [reply]
@ghostdevv this is a good point, and something I thought about how we could solve during the week. I'm wondering if it might be nice to have a binding for `{{ someTable.selectedRow }}`, so you could have a button that you could attach a handler to, and then use all the values in the selected row in your handler. Would this solve your problem?

That way, you wouldn't have to set up row handlers on all your tables. All the selected rows for any table would be available to you at any time.

**Reply by saascrash** [reply]
Binding to the data of other design components would be perfect for dynamic content and much appreciated. This way it makes it incredibly easier to link data from one component to another without having to create data providers and several components just to quickly link text. {{ someTable.selectedRow }} and {{ someTable.selectedRow.columnName }} makes things incredibly quick and easy as well as opens up a ton of customization ability and dashboard possibilities. Basically passing around data through components allows for more functions that can be on a single page and without much configuration.

Good example: I set up a section as Sidebar with Main and put a table in the sidebar. I put a container in the Main with dynamic text based on the selected row. This acts as an enhanced profile view of the selected row and changes as a different row is selected. I can further add action buttons to that container for whatever purpose I see fit and have it pull the data from either the selected row or from {{ someContainer.someField }} depending if there are added fields in the card from another data set.

**Reply by arpit-ma** [reply]
@joebudi the above solution is not working, can you please help me to do this. @ghostdevv @thieverycorp @shogunpurple

**Comment by sergey-dme** [comment]
As the temporary solution to be able to use table row click you can do the following:

1. Add new link component (as a table child)
2. Insert the following Custom CSS to it and adjust the min-width/min-height:

```
position: absolute;
left: 0px; /* Or your margin value to not cover the icon/text in first column */
min-width: 1054px; /* Your table width */
min-height: 55px; /* Your row height */
z-index: 1;
content: "";
``` 
3. Type just a _space_ in Text filed of the component.
4. In Link filed -- Insert the link to view/edit row as you usually use.

**Reply by joebudi** [reply]
This may prove problematic on mobile devices due to the min-width prop.

**Reply by sergey-dme** [reply]
@joebudi You are right Joe, this temporary solution only works for desktop
